src/doc/en/developer/index.rst

```diff
@@ -1,182 +1,186 @@
-  requests, changes in progress, and those already part of Sage
-  today. :ref:`Click here <chapter-sage-trac>` for more information.
+Now here is a brief overview of this guide.
 
-  Importantly, you will need to :ref:`create a trac account
-  <section-trac-account>` in order to contribute.
+- :ref:`section-first-steps`: To share changes with the Sage community, you
+  need to learn about revision control. We use the software Git for this
+  purpose. Here we walk you through from setting up Git on your platform
+  and to preparing a local branch to share with all Sage users.
 
-- **Source code:** You need your own copy of Sage's source code to change it.
-  `Go there <http://doc.sagemath.org/html/en/installation/source.html>`_ to get it
-  and for instructions to build it.
+  .. NOTE::
 
-  If you have never worked on software before, pay close attention to the
-  `prerequisites to compile
-  <http://doc.sagemath.org/html/en/installation/source.html#prerequisites>`_ on your
-  system.
+    As an easy way to get started, you can run and edit Sage's code and contribute
+    your changes using `Gitpod <https://www.gitpod.io>`_, a free online development
+    environment based on VS Code.  It will launch a pre-made workspace with all
+    dependencies and tools installed so that you can start contributing straight
+    away.  Start by `going to Gitpod
+    <https://gitpod.io/#https://github.com/sagemath/sage>`_, and read :ref:`our
+    Gitpod guidelines <section-gitpod>` to learn more.
 
-  As an easy way to get started, you can run and edit Sage's code and contribute
-  your changes using `Gitpod <https://www.gitpod.io>`_,
-  a free online development environment based on VS Code.
-  It will launch a pre-made workspace with all dependencies and tools installed
-  so that you can start contributing straight away.
-  Start by `going to Gitpod <https://gitpod.io/#https://github.com/sagemath/sage>`_,
-  and read :ref:`our Gitpod guidelines <section-gitpod>` to learn more.
+- :ref:`section-development-on-github`: All changes go through `the Sage
+  repository on GitHub <https://github.com/sagemath/sage>`_ at some point. It contains
+  bug reports, enhancement proposals, changes in progress, and indeed all the
+  history of Sage today. You have to be familiar with it to be involved in Sage
+  development.
 
-- **Conventions:** read our :ref:`conventions and guidelines
-  <section-writing-code-for-sage>` for code and documentation.
+- :ref:`section-git-tricks-and-tips`: Here we give an in-depth guide for
+  working with Git for Sage development. Read this when you need help on Git in
+  a tricky situation such as merge conflict.
 
-  For everything related to manuals, tutorials, and languages, :ref:`click here
-  <chapter-sage_manuals>`.
+- :ref:`section-writing-code-for-sage`: This is a guide on conventions in
+  writing code and documentation. To be a good developer, read this part once
+  in a while
```

Comment:
Perhaps: "As conventions evolve over time, also experienced Sage contributors may want to review this section once in a while."

Reply:
Thanks. Done.